docs/error_codes.yaml

```diff
@@ -0,0 +1,147 @@
+#
+# Copyright (C) 2023, Pelican Project, Morgridge Institute for Research
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you
+# may not use this file except in compliance with the License.  You may
+# obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file contains structured documentation about the Pelican parameters.
+# While it is somewhat human-readable, it is meant to help with the documentation
+# generation.
+
+############################
+#    Client Error Codes    #
+############################
+
+############################
+#     Parameter Errors     #
+############################
+---
+type: Parameter
+code: 100
+clientExitCode: 4
+description: >-
+  If the client failed to start, or was started with invalid parameters, or otherwise believes what it
+  was asked to do is impossible or the request itself is malformed.
+retryable: false
+
+############################
+#    Resolution Errors     #
+############################
+---
+type: Resolution
+code: 200
+clientExitCode: 5
+description: >-
+  Indicates that the client failed to even attempt to contact the server
+retryable: false
+
+############################
+#      Contact Errors      #
+############################
+---
+type: Contact
+code: 300
+clientExitCode: 6
+description: >-
+  The client attempted to contact the server at its resolved address and failed to do so
+retryable: false
+---
+type: Contact.Director
+code: 301
+clientExitCode: 6
+description: >-
+  The client attempted to contact the director at its found address but failed to do so
+retryable: false
+---
+type: Contact.Cache
+code: 302
+clientExitCode: 11
+description: >-
+  The client attempted to contact the cache but failed to do so
+retryable: true
+---
+type: Contact.Origin
+code: 303
+clientExitCode: 6
+description: >-
+  The client attempted to contact the origin but failed to do so
+retryable: false
```

Comment:
I guess clients don't really reach out to the registry, but should we think about propagating some kind of error from the Director in the event that the Director can't get something from the registry on behalf of a requested namespace/object?

Reply:
Yeah, I feel this could maybe work for the errors we get that are `no namespace found for path' and such. I'll add this error in. Again, this is a first draft so as I go through and categorize the errors I may find some that I did miss

Reply:
I guess what I said could be a specification error too tho... Either way I will add it in!